src/simulator/Simulator.ts

```diff
@@ -12,20 +13,34 @@ export class Simulator {
     private currentState: WorldState;
     private history: WorldState[];
 
-    constructor(initialState: Scenario, daysPerTurn = 10) {
-        this.scenario = initialState;
+    constructor(scenario: Scenario, daysPerTurn = 10) {
+        this.scenario = scenario;
         this.daysPerTurn = daysPerTurn;
-        this.scaleFactor = initialState.gdpPerDay * 0.2;
+        this.scaleFactor = scenario.gdpPerDay * 0.2;
         this.currentState = this.computeInitialWorldState();
         this.history = [];
     }
 
+    /**
+     * Restarts the scenario at a given turn (turn zero by default).
+     * Returns a new simulator instance.
+     */
+    reset(turn: number = 0): Simulator {
```

Comment:
This is the real critical thing of this PR. The reset method allows the user to specify the turn from which to reset (note that game turn != game days). By default it goes back to turn 0, i.e., starts a new game in the same scenario.